fdc.py

```diff
@@ -0,0 +1,214 @@
+"""
+Handles requests to and responses from
+the USDA FoodData Central (FDC) API.
+"""
+
+import json
+import requests
+from requests.exceptions import RequestException
+
+# the "dataType" parameter is intentionally excluded from the list and search
+# functions as a design decision; the "Branded" will be
+# the only acceptable data type, and will always be used as a filter.
+# The "brandOwner" parameter is also excluded from search, as it doesn't seem like
+# a feature we'll need. This can be easily added if necessary.
+_DATA_TYPES = ["Branded"]
+_FDC_URL = "https://api.nal.usda.gov/fdc/v1"  # base url of the FDC API
+_SECRETS = {"FDC_KEY": None}  # using a dict rather than a global variable
+_SORT_VALUES = {
+    "dataType": "dataType.keyword",
+    "description": "lowercaseDescription.keyword",
+}
+
+
+class FDCResponseList(list):
+    """Wrapper type for adding the success field to response lists."""
+
+    def __getitem__(self, i):
+        if i == "success":
+            return True
+
+        return list.__getitem__(self, i)
+
+    def __contains__(self, o):
+        if o == "success":
+            return True
+
+        return list.__contains__(self, o)
```

Comment:
For readability purposes, what do variables i and o mean in this function? I understand a list is implicitly passed through the parameters for the class FDCResponseList, but what do they mean? Looking over the structure, do I and O mean input and output?

Reply:
Sorry about those unclear names, I used the shortcuts recommended by IntelliSense which inserted the functions' declarations.

`FDCResponseList` is a subclass of `list` which overrides some of its magic methods. The `i` parameter in `__getitem__` is the key used when retrieving a list item using square bracket syntax, e.g. `0` in `response[0]`. This override of `__getitem__` allows `response["success"]` to be valid on response lists. A clearer name for this would be `key`.

In `__contains__`, `o` is the item used in the expression `"value" in response`. That is, `o` will be assigned to `"value"` in the method. I included this so `"success" in response` would be true for response lists. A clearer name for this would be `"item"`.

Both of these always return `True` because the created list is guaranteed to be a valid response object.
I can add another commit changing these parameter names and adding docstrings to make it clear what's happening here.